universum/analyzers/uncrustify.py

```diff
@@ -1,200 +1,202 @@
-
+def _uncrustify_output_parser(files: List[Tuple[Path, Path]],
+                              diff_file_writer: HtmlDiffFileWriter,
+                              _: str) -> List[utils.ReportData]:
+    result: List[utils.ReportData] = []
+    for src_file, uncrustify_file in files:
         with open(src_file) as src:
             src_lines = src.readlines()
         with open(uncrustify_file) as fixed:
             fixed_lines = fixed.readlines()
 
-        file_issues = []
-        matching_blocks = difflib.SequenceMatcher(a=src_lines, b=fixed_lines).get_matching_blocks()
-        previous_match = matching_blocks[0]
-        for match in matching_blocks[1:]:
-            block = get_mismatching_block(previous_match, match, src_lines, fixed_lines)
-            previous_match = match
-            if block:
-                file_issues.append(get_issue_json_format(src_file, block))
-
-        # Generate html diff
-        if file_issues:
-            self.generate_html_diff(uncrustify_file, src_lines, fixed_lines)
-
-        return file_issues
-
-    def execute(self):
-        if not self.settings.cfg_file and ('UNCRUSTIFY_CONFIG' not in os.environ):
-            sys.stderr.write("Please specify the '--cfg_file' parameter "
-                             "or set an env. variable 'UNCRUSTIFY_CONFIG'")
-            return 2
+        issues = _get_issues_from_diff(src_file, src_lines, fixed_lines)
+        if issues:
+            diff_file_writer.write_file(src_file, src_lines, fixed_lines)
+        result.extend(issues)
+    return result
+
+
+def _get_wrapcolumn_tabsize(cfg_file: str) -> Tuple[int, int]:
+    with open(cfg_file) as config:
+        for line in config.readlines():
+            if line.startswith("code_width"):
+                wrapcolumn = int(line.split()[2])
+            if line.startswith("input_tab_size"):
+                tabsize = int(line.split()[2])
+    return wrapcolumn, tabsize
+
+
+def _get_issues_from_diff(src_file: Path, src: List[str], target: List[str]) -> List[utils.ReportData]:
+    result = []
+    matching_blocks: List[difflib.Match] = \
+        difflib.SequenceMatcher(a=src, b=target).get_matching_blocks()
+    previous_match = matching_blocks[0]
+    for match in matching_blocks[1:]:
+        block = _get_mismatching_block(previous_match, match, src, target)
+        previous_match = match
+        if not block:
+            continue
+        line, before, after = block
+        path: Path = src_file.relative_to(Path.cwd())
+        message = _get_issue_message(before, after)
+        result.append(utils.ReportData(
+            symbol="Uncrustify Code Style issue",
+            message=message,
+            path=str(path),
+            line=line
+        ))
+
+    return result
+
+
+def _get_issue_message(before: str, after: str) -> str:
+    # The maximum number of lines to write separate comments for
+    # If exceeded, summarized comment will be provided instead
+    max_lines = 11
```

Comment:
moved inside from global as this value only makes sense within the context of the function